components/prism-turtle.js

```diff
@@ -0,0 +1,33 @@
+Prism.languages.turtle = {
+	'comment': {
+		pattern: /#.*/,
+		greedy: true
+	},
+	'multilineString': {
+		pattern: /(?:(?:"""(?:(?:[^"]|\n|\r)"?"?)*""")|(?:'''(?:(?:[^']|\n|\r)'?'?)*'''))/,
+		greedy: true,
+		alias: "string",
+		inside: {
+			'comment' : {
+				pattern: /#.*/,
+				greedy: true
+			}
+		}
+	},
+	'string': {
+		pattern: /(?:(?:"(?:[^\"\r\n]|\\")*")|(?:'(?:[^\'\r\n]|\\')*'))/,
+		greedy: true
+	},
+	'url': {
+		pattern: /<(?:[^ ])*>/,
+		greedy: true
+	},
+	'function': /\b(?:[^: \r\n]*)?:(?:[^: \r\n,.;\(\)\{\}\[\]\^\.]*)?/,
```

Comment:
`Function` here is a category for a "shortened URL", which has a prefix part, a colon, and a local name. I am open to suggestions for other categories.

Just `:` is intended. It means "prefix with no name" (the part before `:`) and no local name (the part after `:`). this resolves to the URL specified by, e.g. `@prefix : <foo> .`

As to the `\b`, the exact listing of allowed chars is [this](https://www.w3.org/TR/turtle/#grammar-production-PN_CHARS_BASE) - it is a unicode inclusive string.

Reply:
Firstly, `\b` is not Unicode inclusive and secondly, it's causing trouble either way. 
Because of it, the prefix is required. See [example 25](https://www.w3.org/TR/turtle/#h2_sec-examples).

This pattern has to be moved before `keyword` because of the `a`. See example 20.
<details>

![image](https://user-images.githubusercontent.com/20878432/62479652-3dc4a200-b7ae-11e9-8611-53a56e825a91.png)

</details>

More granular tokenization would also be nice. I think that the following as the `inside` of the `function` token should look good.

```js
{
	'prefix': /^[^:]+(?=:)/,
	'punctuation': /:/,
	'name': /[\s\S]+/
}
```

Reply:
I looked a bit at the grammar and found that according to [this rule](https://www.w3.org/TR/turtle/#grammar-production-PN_LOCAL), a local name can contain any number of colons.

Also, I made this pattern for the prefix (and only the prefix) which more closely resembles the actual definition.

```js
prefix = /[A-Z\xC0-\uFFFD](?:[.\-\w\xB7\xC0-\uFFFD]*[-\w\xB7\xC0-\uFFFD])?:/i
```

I'll make one for the local name next.

Reply:
Slightly revised prefix plus full pattern:

```js
prefix = /(?:(?![-._\d\xB7])[-.\w\xB7\xC0-\uFFFD]+)?:/i
localName = /(?![-.])(?:[-.:\w\xC0-\uFFFD]|%[\da-f]{2}|\\.)+/i;
full = /(?:(?![-._\d\xB7])[-.\w\xB7\xC0-\uFFFD]+)?:(?:(?![-.])(?:[-.:\w\xC0-\uFFFD]|%[\da-f]{2}|\\.)+)?/i;
//   = /<prefix>(<localName>)?/
```

Reply:
Fixed

Reply:
Thanks for this. I used it, and reintroduced the punctuation as inside of the prefix.